### PR TITLE
Core/porting omp to parallel utilities 6 mod

### PR DESCRIPTION
**Pull request by ashishdarekar**
**Description**
This PR is to make the modificatons in the code according to the comments given in the PR 8337 [https://github.com/KratosMultiphysics/Kratos/pull/8337]

ParallelUtilities

**Changelog**
-Modification in the SumReduction statement (capturing required data only)

### CODE REVIEW

File: kratos/processes/from_json_check_result_process.h [hunk 1]
```diff
@@ -277,51 +277,54 @@ class KRATOS_API(KRATOS_CORE) FromJSONCheckResultProcess
...
+                return r_check_counter;
             });
         }
 
         for (auto& p_var_array : mpNodalVariableArrayList) {
             const auto& r_var_database = r_node_database.GetVariableData(*p_var_array);
 
-            check_counter = IndexPartition<std::size_t>(r_nodes_array.size()).for_each<SumReduction<IndexType>>([&] (std::size_t Index){
+            check_counter = IndexPartition<std::size_t>(r_nodes_array.size()).for_each<SumReduction<IndexType>>([&it_node_begin, &p_var_array, &r_var_database, &time, this] (std::size_t Index){
+                IndexType r_check_counter = 0;
```
Comment (philbucher) [inline kratos/processes/from_json_check_result_process.h]:
```suggestion
                IndexType counter = 0;
```

the notation `r_` is reserved for references

**Review by philbucher — COMMENTED** [review]
can you please do the same in the ApplyPeriodicConstaintsProcess that you recently modified?

File: kratos/processes/from_json_check_result_process.h (hunk 1 above)
Comment (ashishdarekar) [reply kratos/processes/from_json_check_result_process.h]: Done

**Review by philbucher — APPROVED** [review]
👍

**Pull request merged**